controllers/provider-aws/pkg/apis/aws/validation/infrastructure.go

```diff
@@ -15,13 +15,50 @@
 package validation
 
 import (
+	"fmt"
+
 	apisaws "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/aws"
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
 
 	cidrvalidation "github.com/gardener/gardener/pkg/utils/validation/cidr"
 	apivalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
+// ValidateInfrastructureConfigAgainstCloudProfile validates the given `InfrastructureConfig` against the given `CloudProfile`.
+func ValidateInfrastructureConfigAgainstCloudProfile(infra *apisaws.InfrastructureConfig, shoot *gardencorev1alpha1.Shoot, cloudProfile *gardencorev1alpha1.CloudProfile, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	shootRegion := shoot.Spec.Region
+	for _, region := range cloudProfile.Spec.Regions {
+		if region.Name == shootRegion {
+			allErrs = append(allErrs, validateInfrastructureConfigZones(infra, region.Zones, fldPath.Child("network"))...)
+			break
+		}
+	}
+
+	return allErrs
+}
+
+// validateInfrastructureConfigZones validates the given `InfrastructureConfig` against the given `Zones`.
+func validateInfrastructureConfigZones(infra *apisaws.InfrastructureConfig, zones []gardencorev1alpha1.AvailabilityZone, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	awsZones := sets.NewString()
+	for _, awsZone := range zones {
+		awsZones.Insert(awsZone.Name)
+	}
+
+	for i, zone := range infra.Networks.Zones {
+		if !awsZones.Has(zone.Name) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("zones").Index(i).Child("name"), zone.Name, fmt.Sprintf("supported values: %v", awsZones.UnsortedList())))
+		}
+	}
+
+	return allErrs
+}
+
 // ValidateInfrastructureConfig validates a InfrastructureConfig object.
 func ValidateInfrastructureConfig(infra *apisaws.InfrastructureConfig, nodesCIDR, podsCIDR, servicesCIDR *string) field.ErrorList {
```

Comment:
un-used values `nodesCIDR, podsCIDR, servicesCIDR` ? 
```suggestion
func ValidateInfrastructureConfig(infra *apisaws.InfrastructureConfig) field.ErrorList {
```

Reply:
I think you rather mean `ValidateInfrastructureConfigUpdate `? Thanks for pointing that out, I removed it from there.